src/bot.py

```diff
@@ -75,40 +123,43 @@ def tweet_loop(self, api):
                 else:
                     api.update_status(self.generate_batch())
                     logging.info("Tweeted out a new flower bed!")
-                    logging.info("The next tweet is scheduled to be made in {} minutes".format(cooldown))
+                    logging.info("The next tweet is scheduled to be made in {} seconds".format(cooldown))
                     time.sleep(cooldown)
-            except:
-                print(sys.exc_info())
-                logging.critical("Tweeting failed due to ratelimit. Waiting {} more minutes.".format(cooldown))
-                time.sleep(cooldown)
+                    raise tweepy.TweepError("error")
+            except Exception as ex:
+                exception = type(ex).__name__
                 
+                if exception == 'RateLimitError':
+                    logging.critical("Tweeting failed due to ratelimit. Waiting {} more seconds.".format(cooldown))
+                    time.sleep(cooldown)
+                else:
+                    logging.critical("TweepError not RateLimitError.")

```

Comment:
Then what? Shouldn't the entire bot just crash at this point? That change has no effect.

Reply:
SO i will crash the bot if it's not an authentication error or an rate limit error. Suggestion taken